--- sysdeps/generic/dl-cache.c~	Fri Dec 27 06:52:07 1996
+++ sysdeps/generic/dl-cache.c	Thu Feb 27 19:54:52 1997
@@ -73,7 +73,8 @@
     }
 
   for (i = 0; i < cache->nlibs; ++i)
-    if (cache->libs[i].flags == 1 && /* ELF library entry.  */
+    if ((cache->libs[i].flags == 1 ||
+	 cache->libs[i].flags == 3) && /* ELF library entry.  */
 	/* Make sure string table indices are not bogus before using them.  */
 	cache->libs[i].key < cachesize - sizeof *cache &&
 	cache->libs[i].value < cachesize - sizeof *cache &&
